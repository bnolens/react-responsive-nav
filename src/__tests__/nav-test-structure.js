import React from 'react'
import ReactDOM from 'react-dom'
import Nav from '../nav.jsx'
import renderer from 'react/lib/ReactTestRenderer'

console.error = jest.genMockFn()

describe('Nav', () => {

  it('displays nav links when mobileMenuExpand is clicked', () => {
    const component = renderer.create(
      <Nav onClick={(button) => { return; }}
           onSearch={(event) => { return; }}
           menu={[
            {id:"settings", label:"Settings", items:[
              {id:"logout", label:"Logout", href:"#"}
            ]}
      ]}></Nav>
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()

    const mobileMenuExpand = tree.children[0].children[0]
    mobileMenuExpand.props.onClick()

    tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('displays the search box and closes the nav links when mobileMenuExpand is clicked', () => {
    const component = renderer.create(
      <Nav onClick={(button) => { return; }}
           onSearch={(event) => { return; }}
           menu={[
            {id:"settings", label:"Settings", items:[
              {id:"logout", label:"Logout", href:"#"}
            ]}
      ]}></Nav>
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()

    const mobileSearchExpand = tree.children[0].children[2]
    mobileSearchExpand.props.onClick()

    tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
