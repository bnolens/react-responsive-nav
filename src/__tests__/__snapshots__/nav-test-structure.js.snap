exports[`Nav displays nav links when mobileMenuExpand is clicked 1`] = `
<nav>
  <div
    className="inner">
    <div
      className=""
      id="mobile-menu-expand"
      onClick={[Function toggleMenuAction]}>
      <div
        className="bar" />
      <div
        className="bar" />
      <div
        className="bar" />
    </div>
    <a
      id="logo"
      onClick={[Function clickButtonAction]} />
    <div
      className=""
      id="search-expand-mobile"
      onClick={[Function toggleSearchAction]}>
      <div
        className="icon" />
    </div>
    <div
      className=""
      id="nav-links">
      <span
        className="dropdownContainer">
        <a
          className="button dropdown"
          href={undefined}
          id="settings"
          label="Settings"
          onBlur={[Function blurAction]}
          onFocus={[Function focusDropdownAction]}
          onMouseDown={[Function collapseDropdownIfExpanded]}
          tabIndex={0}>
          Settings
           ▼
        </a>
        <div
          className="sublinks">
          <a
            className="button"
            href="#"
            id="logout"
            onBlur={[Function blurAction]}
            onClick={[Function onClick]}
            onFocus={[Function focusButtonAction]}
            tabIndex={0}>
            Logout
          </a>
        </div>
      </span>
    </div>
    <div
      className=""
      id="search">
      <input
        id="search-input"
        onChange={[Function searchChange]}
        placeholder="Search..."
        type="text" />
    </div>
  </div>
</nav>
`;

exports[`Nav displays nav links when mobileMenuExpand is clicked 2`] = `
<nav>
  <div
    className="inner">
    <div
      className="expanded"
      id="mobile-menu-expand"
      onClick={[Function toggleMenuAction]}>
      <div
        className="bar" />
      <div
        className="bar" />
      <div
        className="bar" />
    </div>
    <a
      id="logo"
      onClick={[Function clickButtonAction]} />
    <div
      className=""
      id="search-expand-mobile"
      onClick={[Function toggleSearchAction]}>
      <div
        className="icon" />
    </div>
    <div
      className="expanded"
      id="nav-links">
      <span
        className="dropdownContainer">
        <a
          className="button dropdown"
          href={undefined}
          id="settings"
          label="Settings"
          onBlur={[Function blurAction]}
          onFocus={[Function focusDropdownAction]}
          onMouseDown={[Function collapseDropdownIfExpanded]}
          tabIndex={0}>
          Settings
           ▼
        </a>
        <div
          className="sublinks">
          <a
            className="button"
            href="#"
            id="logout"
            onBlur={[Function blurAction]}
            onClick={[Function onClick]}
            onFocus={[Function focusButtonAction]}
            tabIndex={0}>
            Logout
          </a>
        </div>
      </span>
    </div>
    <div
      className=""
      id="search">
      <input
        id="search-input"
        onChange={[Function searchChange]}
        placeholder="Search..."
        type="text" />
    </div>
  </div>
</nav>
`;

exports[`Nav displays the search box and closes the nav links when mobileMenuExpand is clicked 1`] = `
<nav>
  <div
    className="inner">
    <div
      className=""
      id="mobile-menu-expand"
      onClick={[Function toggleMenuAction]}>
      <div
        className="bar" />
      <div
        className="bar" />
      <div
        className="bar" />
    </div>
    <a
      id="logo"
      onClick={[Function clickButtonAction]} />
    <div
      className=""
      id="search-expand-mobile"
      onClick={[Function toggleSearchAction]}>
      <div
        className="icon" />
    </div>
    <div
      className=""
      id="nav-links">
      <span
        className="dropdownContainer">
        <a
          className="button dropdown"
          href={undefined}
          id="settings"
          label="Settings"
          onBlur={[Function blurAction]}
          onFocus={[Function focusDropdownAction]}
          onMouseDown={[Function collapseDropdownIfExpanded]}
          tabIndex={0}>
          Settings
           ▼
        </a>
        <div
          className="sublinks">
          <a
            className="button"
            href="#"
            id="logout"
            onBlur={[Function blurAction]}
            onClick={[Function onClick]}
            onFocus={[Function focusButtonAction]}
            tabIndex={0}>
            Logout
          </a>
        </div>
      </span>
    </div>
    <div
      className=""
      id="search">
      <input
        id="search-input"
        onChange={[Function searchChange]}
        placeholder="Search..."
        type="text" />
    </div>
  </div>
</nav>
`;

exports[`Nav displays the search box and closes the nav links when mobileMenuExpand is clicked 2`] = `
<nav>
  <div
    className="inner">
    <div
      className=""
      id="mobile-menu-expand"
      onClick={[Function toggleMenuAction]}>
      <div
        className="bar" />
      <div
        className="bar" />
      <div
        className="bar" />
    </div>
    <a
      id="logo"
      onClick={[Function clickButtonAction]} />
    <div
      className="expanded"
      id="search-expand-mobile"
      onClick={[Function toggleSearchAction]}>
      <div
        className="icon" />
    </div>
    <div
      className=""
      id="nav-links">
      <span
        className="dropdownContainer">
        <a
          className="button dropdown"
          href={undefined}
          id="settings"
          label="Settings"
          onBlur={[Function blurAction]}
          onFocus={[Function focusDropdownAction]}
          onMouseDown={[Function collapseDropdownIfExpanded]}
          tabIndex={0}>
          Settings
           ▼
        </a>
        <div
          className="sublinks">
          <a
            className="button"
            href="#"
            id="logout"
            onBlur={[Function blurAction]}
            onClick={[Function onClick]}
            onFocus={[Function focusButtonAction]}
            tabIndex={0}>
            Logout
          </a>
        </div>
      </span>
    </div>
    <div
      className="expanded"
      id="search">
      <input
        id="search-input"
        onChange={[Function searchChange]}
        placeholder="Search..."
        type="text" />
    </div>
  </div>
</nav>
`;
